File: server/apps/nexotype/schemas/user/recommendation_schemas.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

# ==========================================
# Request Schemas
# ==========================================

class RecommendationCreate(BaseModel):
    """Schema for creating a new recommendation"""
    user_profile_id: int
    asset_id: int
    reason: str
    priority: str


class RecommendationUpdate(BaseModel):
    """Schema for updating a recommendation"""
    user_profile_id: int = None
    asset_id: int = None
    reason: str = None
    priority: str = None


# ==========================================
# Response Schemas
# ==========================================

class RecommendationDetail(BaseModel):
    """Schema for recommendation details"""
    id: int
    user_profile_id: int
    asset_id: int
    reason: str
    priority: str
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class RecommendationListResponse(BaseModel):
    """Response schema for listing recommendations"""
    success: bool
    data: list[RecommendationDetail] | None = None
    count: int = 0
    error: str | None = None


class RecommendationResponse(BaseModel):
    """Response schema for single recommendation operations"""
    success: bool
    data: RecommendationDetail | None = None
    error: str | None = None


class MessageResponse(BaseModel):
    """Simple message response"""
    success: bool
    message: str | None = None
    error: str | None = None

File: client/src/modules/nexotype/schemas/user/recommendation.schemas.ts
import { z } from 'zod';

// ==========================================
// Enums
// ==========================================

export const PriorityEnum = z.enum(['High', 'Medium', 'Low']);

// ==========================================
// Options (for Select dropdowns)
// ==========================================

/** Recommendation.priority */
export const PRIORITY_OPTIONS = [
  { label: 'High', value: 'High' },
  { label: 'Medium', value: 'Medium' },
  { label: 'Low', value: 'Low' },
] as const;

// ==========================================
// Full Schema (GET responses)
// ==========================================

export const RecommendationSchema = z.object({
  id: z.number(),
  user_profile_id: z.number().int(),
  asset_id: z.number().int(),
  reason: z.string().min(1),
  priority: PriorityEnum,
  created_at: z.string(),
  updated_at: z.string().nullable(),
});

// ==========================================
// Input Schemas (POST/PUT)
// ==========================================

export const CreateRecommendationSchema = z.object({
  user_profile_id: z.number().int(),
  asset_id: z.number().int(),
  reason: z.string().min(1),
  priority: PriorityEnum,
});

export const UpdateRecommendationSchema = z.object({
  user_profile_id: z.number().int().optional(),
  asset_id: z.number().int().optional(),
  reason: z.string().min(1).optional(),
  priority: PriorityEnum.optional(),
});

// ==========================================
// Types
// ==========================================

export type Recommendation = z.infer<typeof RecommendationSchema>;
export type CreateRecommendation = z.infer<typeof CreateRecommendationSchema>;
export type UpdateRecommendation = z.infer<typeof UpdateRecommendationSchema>;
export type Priority = z.infer<typeof PriorityEnum>;

// ==========================================
// Response Types
// ==========================================

export type RecommendationResponse = {
  success: boolean;
  data?: Recommendation;
  error?: string;
};

export type RecommendationsResponse = {
  success: boolean;
  data?: Recommendation[];
  count?: number;
  error?: string;
};
